apollo-client/src/main/java/com/ctrip/framework/apollo/internals/AbstractConfig.java

```diff
@@ -492,19 +494,24 @@ private List<ConfigChangeListener> findMatchedConfigChangeListeners(Set<String>
   }
 
   private void notifyAsync(final ConfigChangeListener listener, final ConfigChangeEvent changeEvent) {
+    CountDownLatch latch = new CountDownLatch(1);
     m_executorService.submit(new Runnable() {
       @Override
       public void run() {
         String listenerName = listener.getClass().getName();
         Transaction transaction = Tracer.newTransaction("Apollo.ConfigChangeListener", listenerName);
         try {
+          if(!(listener instanceof AutoUpdateConfigChangeListener)){
```

Comment:
The AutoUpdateConfigChangeListener is only triggered in a Spring environment, however, apollo should run in all environments.